ui/components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal.tsx

```diff
@@ -193,29 +167,125 @@ export function AssetPickerModal({
   }, []);
 
   // filter and exclude ERC1155
-  const collectionDataFiltered = (collectionsData as Collection[])
-    .map((collection) => {
-      return {
-        ...collection,
-        nfts: collection.nfts.filter(
-          (nft) => nft.standard !== TokenStandard.ERC1155,
-        ),
-      };
-    })
-    .filter((collection) => collection.nfts.length > 0);
+  const collectionDataFiltered = (collectionsData as Collection[]).filter(
+    (collection) => collection.nfts.length > 0,
+  );
 
   const handleSearch = (query: string) => {
     setSearchQuery(query);
   };
 
   const handleAssetChange = (token: Token) => () => {
     onAssetChange(token);
-    setSelectedToken(token.address);
     onClose();
   };
 
   const defaultActiveTabKey = asset?.type === AssetType.NFT ? 'nfts' : 'tokens';
 
+  const AssetList = useCallback(() => {
+    const tokenList = tokensWithBalances.map((token: Token) => {
+      token.isSelected =
+        token.address?.toLowerCase() === selectedToken?.toLowerCase();
+      return token;
+    });
+
+    tokenList.push({
+      address: null,
+      symbol: nativeCurrency,
+      decimals: 18,
+      image: nativeCurrencyImage,
+      balance: balanceValue,
+      string: primaryCurrencyProperties.value,
+      type: AssetType.native,
+      isSelected: !selectedToken,
+    });
+
+    tokenList.sort((a, b) => {
+      if (a.type === AssetType.native) {
+        return -1;
+      } else if (b.type === AssetType.native) {
+        return 1;
+      }
+      return 0;
+    });
```

Comment:
Does this just move the native token to index 0? Why not just `unshift` instead of `push` above?

Reply:
You're right – I didn't write it lol